javaslang/src/main/java/javaslang/Value.java

```diff
@@ -516,7 +516,7 @@ default CharSeq toCharSeq() {
      */
     @SuppressWarnings("unchecked")
     @GwtIncompatible("reflection is not supported")
-    default T[] toJavaArray(Class<T> componentType) {
+    default T[] toJavaArray(Class<? super T> componentType) {
```

Comment:
As @paplorinc said, these are our options:

```java
// unsafe
T[] toJavaArray(Class<? super T> componentType) // one size fits all

vs

// safe
T[] toJavaArray(Class<T> componentType) // variant for non-generic types
T[] toJavaArray(T[] array)
```

I think the first, unsafe variant is no viable solution.

Example:

```java
// before
Value<CharSequence> v = Option.of("");
CharSequence[] s = v.toJavaArray(CharSequence.class);

// after
Value<String> v = Option.of("");
String[] s = v.toJavaArray(CharSequence.class); // will eventually throw at runtime
```

The method `toJavaArray(Class<T>)` is still useful for non-generic types,
we do not need to deprecate it.